fluentd/configs.d/filter-events-multiline.conf

```diff
@@ -0,0 +1,33 @@
+# Aggregate multiline log messages originating from containers.
+
+# Since Fluentd 0.12 does not support nested attributes easily (see
+# <https://github.com/fluent/fluent-plugin-rewrite-tag-filter/pull/42#issuecomment-337088353>),
+# use the Fluentd record transformer plugin to extract the Docker container ID
+# and the Kubernetes container name.
+<filter kubernetes.**>
+  @type record_transformer
+  enable_ruby
+  <record>
+    containerid "${record['docker']['container_id']}"
```

Comment:
are 'docker' and 'kubernetes' always present? if so what if 'container_id' and 'container_name' are missing?

Reply:
>  Additionally, do we know if there is any memory or other performance related impact to introducing this change?

I can guarantee there is an impact, both when there are no multiline logs, and when there is, I have no idea what it will be.

> are 'docker' and 'kubernetes' always present?

No, in error cases, depending on how the fluent-plugin-kubernetes handles errors